apps/backend/src/geolocation/geolocation.controller.ts

```diff
@@ -0,0 +1,12 @@
+import { Controller, Get, Query } from '@nestjs/common';
+import { OpenCageService } from '../frameworks/opencage.service';
+
+@Controller('api/v1/location')
+export class GeolocationController {
+  constructor(private readonly geolocationService: OpenCageService) {}
```

Comment:
the controller shouldn't know about which service we're using. instead, we should only depend on the abstract class you created.